4.5_dry_up_with_inheritance/src/main/java/com/thinkful/drills/Character.java

```diff
@@ -0,0 +1,93 @@
+package com.thinkful.drills;
+
+public class Character {
+  private String name;
+  private String description;
+  private int health;
+  private double strength;
+
+  public static final int INITIAL_HEALTH = 100;
+  public static final double INITIAL_STRENGTH = 100.0;
+
+  /**
+   * Create a Character with all default values.
+   */
+  public Character() {
+    this("");
+  }
+
+  /**
+   * Create a Character with name and default values.
+   */
+  public Character(String name) {
+    this(name, "");
+  }
+
+  /**
+   * Create a Character with name, description and default values.
+   */
+  public Character(String name, String description) {
+    this(name, description, INITIAL_HEALTH);
+  }
+
+  /**
+   * Create a Character with name, description, health and default values.
+   */
+  public Character(String name, String description, int health) {
+    this(name, description, health, INITIAL_STRENGTH);
+  }
+
+  /**
+   * Create a Character with name, description, health and strength.
+   */
+  public Character(String name, String description, int health, double strength) {
+    this.setName(name);
+    this.setHealth(health);
+    this.setStrength(strength);
+    this.setDescription(description);
+  }
```

Comment:
As part of the DRY principle, I would send this code back to a jr dev if it was submitted to me to review. When there are setters for attributes the constructor really should only be for required fields that do not have a default value. Basically cleaning up the constructors would be another good exercise for the DRY principle. I think the best constructor for this class would be 
`public Character(String name, String description) {
    this.setName(name);
    this.setDescription(description);
    this.setHealth(INITIAL_HEALTH);
    this.setStrength(INITIAL_STRENGTH);
  }`
This appropriately defines what is actually needed when creating a new Character.